Document/0x05g-Testing-Network-Communication.md

```diff
@@ -239,6 +241,48 @@ myWebView.setWebViewClient(new WebViewClient(){
 });
 ```
 
+##### Xamarin Applications
+
+Applications developed in Xamarin will typically use ServicePointManager to implement pinning.
+
+Normally a function is created to check the certificate(s) and return the boolean value to the method ServerCertificateValidationCallback:
+
+```c#
+[Activity(Label = "XamarinPinning", MainLauncher = true)]
+    public class MainActivity : Activity
+    {
+
+        private const string SupportedPublicKey = "3082010A02820101009CD30CF05AE52E47B7725D3783B3686330EAD735261925E1BDBE35F170922FB7B84B4105ABA99E350858ECB12AC468870BA3E375E4E6F3A76271BA7981601FD7919A9FF3D0786771C8690E9591CFFEE699E9603C48CC7ECA4D7712249D471B5AEBB9EC1E37001C9CAC7BA705EACE4AEBBD41E53698B9CBFD6D3C9668DF232A42900C867467C87FA59AB8526114133F65E98287CBDBFA0E56F68689F3853F9786AFB0DC1AEF6B0D95167DC42BA065B299043675806BAC4AF31B9049782FA2964F2A20252904C674C0D031CD8F31389516BAA833B843F1B11FC3307FA27931133D2D36F8E3FCF2336AB93931C5AFC48D0D1D641633AAFA8429B6D40BC0D87DC3930203010001";
```

Comment:
What type of string is this? What does it acecpt? (E.g. is this a pem encoded cert of a der encoded public key?)

Reply:
Added some comments and debug code that allows to understand the value is and how can be obtained.